fe/fe-core/src/main/java/org/apache/doris/datasource/ExternalCatalog.java

```diff
@@ -258,6 +264,38 @@ public boolean ifNotSetFallbackToSimpleAuth() {
     // Will be called when creating catalog(not replaying).
     // Subclass can override this method to do some check when creating catalog.
     public void checkWhenCreating() throws DdlException {
+        boolean testConnection = Boolean.parseBoolean(
+                catalogProperty.getOrDefault(TEST_CONNECTION, String.valueOf(DEFAULT_TEST_CONNECTION)));
+
+        if (!testConnection) {
+            return;
+        }
+
+        MetaConnectivityTester metaTester = catalogProperty.getMetastoreProperties().createConnectivityTester();
+        try {
+            metaTester.testConnection();
+        } catch (Exception e) {
+            throw new DdlException(metaTester.getTestType() + " connectivity test failed: " + e.getMessage());
+        }
+
+        String testLocation = metaTester.getTestLocation();
```

Comment:
There will be multi storage properties, but only one test location here.
So you may use wrong storage property to test the location, which will throw exception.
So maybe for multi storage properties, we can let test pass only if at least one storage test is passed